src/aequitas/plotting.py

```diff
@@ -37,31 +36,45 @@ def assemble_ref_groups(disparities_table, ref_group_flag='_ref_group_value',
     """
     ref_groups = {}
     ref_group_cols = \
-        list(disparities_table.columns[disparities_table.columns.str.contains(
+        set(disparities_table.columns[disparities_table.columns.str.contains(
             ref_group_flag)])
 
+    # Note: specific measures is a set
     if specific_measures:
-        ref_group_cols = \
-            [measure + ref_group_flag for measure in specific_measures if
-             measure + ref_group_flag in ref_group_cols]
+        specific_measures.intersection({label_score_ref})
+        ref_group_cols = {measure + ref_group_flag for measure in specific_measures if
+             measure + ref_group_flag in ref_group_cols}
 
     attributes = list(disparities_table.attribute_name.unique())
+
     for attribute in attributes:
         attr_table = \
             disparities_table.loc[disparities_table['attribute_name'] == attribute]
+
         attr_refs = {}
         for col in ref_group_cols:
+            if col in ('label' + ref_group_flag, 'score' + ref_group_flag):
+                continue
+
             metric_key = "".join(col.split(ref_group_flag))
             attr_refs[metric_key] = \
                 attr_table.loc[attr_table['attribute_name'] == attribute, col].min()
         if label_score_ref:
-            if label_score_ref + ref_group_flag in ref_group_cols:
-                attr_refs['label_value'] = attr_refs[label_score_ref]
-                attr_refs['score'] = attr_refs[label_score_ref]
-            else:
-                raise ValueError("The specified reference measure for label"
-                                 " value and score is not included in the "
-                                 "data frame.")
+            tried = 0
+
+            if label_score_ref + ref_group_flag not in disparities_table.columns:
+                while (label_score_ref + ref_group_flag not in disparities_table.columns) and (tried <= len(specific_measures)):
+                    label_score_ref = next(iter(specific_measures))
```

Comment:
is this expected to get a different element from `specific_measures` each time? it just doesn't really seem like it would. (I don't see it getting mutated.)

```python
>>> s = set(range(5))
>>> for _count in range(3):
    print(next(iter(s)))
0
0
0
```

…is there any point to `tried` beyond advancing through `specific_measures`? perhaps:

```python
is_valid_label_ref = lambda label: label + ref_group_flag in disparities_table.columns

if not is_valid_label_ref(label_score_ref):
    try:
        label_score_ref = next(measure for measure in specific_measures if is_valid_label_ref(measure))
    except StopIteration:
        pass  # or perhaps raise an error or something since nothing was valid

    logging.warning(…)
```